api: make tests use immediate_stream

ddowl opened the pull request:
Most tests should use immediate_stream as the default packet_generator
to ensure that they are _speedy_.

This change doesn't prune`LinearIntervalGenerator` or `ExhaustiveDrain` since they may still be useful.

fixes #31

ddowl committed:
api: make tests use immediate_stream

Most tests should use immediate_stream as the default packet_generator
to ensure that they are _speedy_.

scgruber reviewed (commented):
I don't understand why we're keeping tests that have waits in them? This goes against the general testing principle of not sleeping or waiting, so I'd rather we didn't.

ddowl commented:
I'm including them so we can be sure that each type of element can deal with bursty streams of packets, and handles the `Some(Async::NotReady)` case. It's the easiest way to invoke `Some(Async::NotReady)` from the packet flow, since a direct stream (where each call to `poll` yields a packet), always returns `Some(Async::Ready)` until the stream ends, and returns `None`. Introducing a bursty packet flow can also reveal weird synchronization issues between `Future`s.

Its true that it makes the test slower, but unfortunately there's not a great way to ensure that `poll` is properly tested otherwise. We can always reduce the wait interval, but then we wouldn't want to lower it so much that the `NotReady` case never gets invoked.

ddowl commented:
If we can think of a faster way to test the `NotReady` cases, we can reserve `PacketIntervalGenerator` and variants for slower integration tests.

CollinValley reviewed (commented):
I approve of this, let's reduce the interval to 10ms instead of 100, accomplishes the same thing, but it means the tests will complete a lot faster. Cargo test is multi-threaded so if we keep everything resolving in under a second that would be quick enough.

ddowl committed:
test: reduce PacketIntervalGenerator wait time

from 100ms to 10ms

--From Collin
Also fix the overseer being added to the new tests

ddowl commented:
Updated wait times.

CollinValley reviewed (dismissed):
Looks good

CollinValley commented:
Merged this PR against master, cleanup up some small build errors.
